client/reader/search-stream/suggestion-provider.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { map, sampleSize } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import TagSubscriptions from 'lib/reader-tags/subscriptions';
+import i18nUtils from 'lib/i18n-utils';
+import { suggestions } from 'reader/search-stream/suggestions';
+
+export default ( Element, count = 3 ) => {
+	return class SuggestionsListener extends React.Component {
+
+		state = {
+			suggestions: this.getSuggestions()
+		}
+
+		componentWillMount() {
+			TagSubscriptions.on( 'change', this.handleChange );
+		}
+
+		componentWillUnmount() {
+			TagSubscriptions.off( 'change', this.handleChange );
+		}
+
+		suggestionsFromTags() {
+			const tags = TagSubscriptions.get();
+			if ( tags ) {
+				if ( tags.length <= count ) {
+					return [];
+				}
+				return map( sampleSize( tags, count ), tag => ( tag.display_name || tag.slug ).replace( /-/g, ' ' ) );
```

Comment:
i think it may be hard-coded to the slug? I'm mostly matching how we show them in the sidebar.